plugins/src/main/java/com/google/firebase/gradle/plugins/GradleExtensions.kt

```diff
@@ -186,4 +192,53 @@ inline fun <reified T : Any> org.gradle.api.plugins.PluginManager.`apply`(): Uni
  * ```
  */
 val Dependency.artifactName: String
-  get() = listOf(group, name, version).filterNotNull().joinToString(":")
+  get() = listOfNotNull(group, name, version).joinToString(":")
+
+/**
+ * Creates an archive of this directory at the [dest] file.
+ *
+ * Can only be ran within the context of a [Task], as outside of a [Task] so you should likely be
+ * using the `copy` or `sync` tasks instead.
+ */
+context(Task)
+fun File.zipFilesTo(dest: File): File {
+  logger.info("Zipping '$absolutePath' to '${dest.absolutePath}'")
+
+  logger.debug("Ensuring parent directories are present for zip file")
+  dest.parentFile?.mkdirs()
+
+  logger.debug("Creating empty zip file to write to")
```

Comment:
Is this necessary? It would be very unlikely that creating the new file fails, rigth?

Reply:
Actually, I added this _because_ it happened while I was writing it. Depending on your file system perms regarding temp dirs, the gradle runner might not have sufficient perms. This is something you'd have to fix as a developer- but it's something that deff can happen lol